### PR TITLE
glob: use ghc option

### PR DESCRIPTION
**Pull request by Chi-EEE**

### CODE REVIEW

**Comment by Chi-EEE** [comment]
Links are not found but the config by default is no headeronly.
https://github.com/xmake-io/xmake-repo/actions/runs/10673054613/job/29581543048?pr=5118
```
D:\a\xmake-repo\xmake-repo\xmakesrc\xmake\xmake.exe install -y --nopkgs -o C:\Users\runneradmin\AppData\Local\.xmake\packages\g\glob\2024.04.18\b56e77bdd7ae4fc094bffbedadb60b2b --diagnosis --verbose --yes
installing glob ..
installing glob to C:\Users\runneradmin\AppData\Local\.xmake\packages\g\glob\2024.04.18\b56e77bdd7ae4fc094bffbedadb60b2b ..
> copy build_b56e77bd\windows\x64\release\glob.dll to C:\Users\runneradmin\AppData\Local\.xmake\packages\g\glob\2024.04.18\b56e77bdd7ae4fc094bffbedadb60b2b\bin
> copy include\glob\glob.h to C:\Users\runneradmin\AppData\Local\.xmake\packages\g\glob\2024.04.18\b56e77bdd7ae4fc094bffbedadb60b2b\include\glob\glob.h
install ok!
finding glob from xmake ..
checking for xmake::glob ... glob 2024.04.18
{ 
  license = "MIT",
  sysincludedirs = { 
    "C:\Users\runneradmin\AppData\Local\.xmake\packages\g\glob\2024.04.18\b56e77bdd7ae4fc094bffbedadb60b2b\include", 
  },
  shared = true,
  version = "2024.04.18", 
}

patching C:\Users\runneradmin\AppData\Local\.xmake\packages\g\glob\2024.04.18\b56e77bdd7ae4fc094bffbedadb60b2b\lib\pkgconfig\glob.pc ..
checking for cl.exe ... C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\bin\HostX64\x64\cl.exe
checking for the c++ compiler (cxx) ... cl.exe
checking for the linker (ld) ... link.exe
checking for link.exe ... C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\bin\HostX64\x64\link.exe
checking for the shared library linker (sh) ... link.exe
error: @programdir\core\package\package.lua:2444: package(glob): links not found!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1003]:
    [@programdir\core\package\package.lua:2444]: in function '_generate_build_configs'
    [@programdir\core\package\package.lua:2608]: in function 'check_cxxsnippets'
    [D:\a\xmake-repo\xmake-repo\packages\g\glob\xmake.lua:53]: in function 'script'
    [...dir\modules\private\action\require\impl\utils\filter.lua:114]: in function 'call'
    [...dir\modules\private\action\require\impl\actions\test.lua:41]:
    [...\modules\private\action\require\impl\actions\install.lua:432]:

  => install glob 2024.04.18 .. failed
error: @programdir\core\main.lua:329: @programdir\modules\async\runjobs.lua:325: ...\modules\private\action\require\impl\actions\install.lua:494: install failed!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1003]:
    [...\modules\private\action\require\impl\actions\install.lua:494]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [...\modules\private\action\require\impl\actions\install.lua:361]:
    [...modules\private\action\require\impl\install_packages.lua:496]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:
```

**Comment by star-hengxing** [comment]
Missing `add_rules("utils.symbols.export_all")`

File: packages/g/glob/xmake.lua [hunk 1]
```diff
@@ -13,29 +13,39 @@ package("glob")
...
     on_install(function (package)
-        if package:config("ghc_filesystem") then
-            package:add("defines", "GLOB_USE_GHC_FILESYSTEM")
-        end
         if package:config("header_only") then
             os.cp("single_include/glob", package:installdir("include"))
         else
+            local configs = {}
+            if package:config("ghc_filesystem") then
+                package:add("defines", "GLOB_USE_GHC_FILESYSTEM")
```
Comment (star-hengxing) [inline packages/g/glob/xmake.lua:29]: Already add defines in on_load

File: packages/g/glob/xmake.lua [hunk 2]
```diff
@@ -13,29 +13,39 @@ package("glob")
...
+                option("ghc_filesystem", {default = false, showmenu = true, description = "Use ghc_filesystem"})
+                if has_config("ghc_filesystem") then
+                    add_requires("ghc_filesystem")
+                end
                 target("glob")
                     set_kind("$(kind)")
                     set_languages("cxx17")
+                    if has_config("ghc_filesystem") then
+                        add_packages("ghc_filesystem")
+                    end
```
Comment (star-hengxing) [inline packages/g/glob/xmake.lua:43]: Maybe need to add defines `GLOB_USE_GHC_FILESYSTEM`?

**Pull request merged**